core-tests/shared/src/test/scala/zio/ZIOSpec.scala

```diff
@@ -786,25 +786,65 @@ object ZIOSpec extends ZIOBaseSpec {
       } @@ nonFlaky
     ),
     suite("partitionM")(
-      testM("collect only successes") {
+      testM("collects only successes") {
         val in = List.range(0, 10)
         for {
-          res <- ZIO.partitionM(in)(a => ZIO.effect(a))
+          res <- ZIO.partitionM(in)(a => ZIO.succeed(a))
```

Comment:
I noticed at some point we did a global import of `zio.CanFail.canFail` at the top of this file. Can we please remove that and do imports for specific tests when necessary? Using that import essentially removes the functionality of `CanFail`, so we need to use it sometimes for specific tests but generally we want to keep that functionality so the compiler helps flag for us things that aren't making sense.